3 -  Start Trip Page

jaygiang opened the pull request:
issue #3 

- Added form inputs to page
- Created css file for Start Trip page
- Created form validation utils 
  - Checks range for latitude and longitude


Form submission:
![Mar-15-2025 01-08-12](https://github.com/user-attachments/assets/19c797d7-ac79-4d20-ba3f-bc3bc29c1e9b)

Form validation:
![Mar-15-2025 01-08-05](https://github.com/user-attachments/assets/9cbafcad-03be-414c-91ba-320bf6f02b72)

thgaskell commented on src/utils/validation.js:
```diff
@@ -0,0 +1,81 @@
+ * @returns {string|null} Error message or null if valid
+ */
+export const validateRequired = (value, fieldName) => {
+  if (!value && value !== 0) {
```
This may be better if it were checking if the value were something like: `null`, `undefined`, or `''`.

I could see `0` being a valid value for a required field.

jaygiang commented:
- Implemented `validateChain` utility that streamlines form field validation by allowing sequential validation checks
- Improved required field validation to handle edge cases (like the value `0`)
- Added precise input control for coordinate fields:
- Updated form container width to match USWDS design patterns (20rem on desktop)
- Centralized field name constants to reduce duplication

![Mar-21-2025 13-10-35](https://github.com/user-attachments/assets/e3769b9b-00dc-4f68-b68a-cae02594408c)

jaygiang commented:
Padding `larger` property in USWDS `Form` component makes form width larger on desktop view

![Screenshot 2025-03-23 at 4 00 49 PM](https://github.com/user-attachments/assets/4d8b6cff-9a24-4358-82ea-8aeef7456a34)

thgaskell commented:
Merged in #15.
